client/my-sites/post-type-list/index.jsx

```diff
@@ -33,124 +25,145 @@ import PostTypeListEmptyContent from './empty-content';
 		wrapTitles: PropTypes.bool,
+		scrollContainer: PropTypes.object,
 
 		// Connected props
 		siteId: PropTypes.number,
-		lastPage: PropTypes.number,
 		posts: PropTypes.array,
-		requestingLastPage: PropTypes.bool,
+		isRequestingPosts: PropTypes.bool,
+		lastPage: PropTypes.number,
+		isRequestingLastPage: PropTypes.bool,
 	};
 
 	constructor() {
 		super( ...arguments );
 
-		this.renderPostRow = this.renderPostRow.bind( this );
-		this.cellRendererWrapper = this.cellRendererWrapper.bind( this );
+		this.renderPost = this.renderPost.bind( this );
 		this.renderPlaceholder = this.renderPlaceholder.bind( this );
-		this.setRequestedPages = this.setRequestedPages.bind( this );
-		this.setListRef = this.setListRef.bind( this );
-		this.handleHeightChange = this.handleHeightChange.bind( this );
-		this.getPostRowHeight = this.getPostRowHeight.bind( this );
 
-		this.rowHeights = {};
+		this.maybeLoadNextPage = this.maybeLoadNextPage.bind( this );
+		this.scrollListener = throttle( this.maybeLoadNextPage, 100 );
+		window.addEventListener( 'scroll', this.scrollListener );
 
+		const maxRequestedPage = this.estimatePageCountFromPosts( this.props.posts );
 		this.state = {
-			requestedPages: this.getInitialRequestedPages( this.props )
+			maxRequestedPage,
 		};
-	}
-
-	componentWillMount() {
-		// NOTE: Assumes that this property does not change for a given
-		// instance of this component
-		this.defaultPostRowHeight = this.props.largeTitles
-			? DEFAULT_POST_ROW_HEIGHT_LARGE
-			: DEFAULT_POST_ROW_HEIGHT_NORMAL;
-	}
-
-	componentDidMount() {
-		if ( this.props.wrapTitles ) {
-			// Note: Assumes that this property does not change
-			this.resizeListener = throttle( this.handleWindowResize, 50 );
-			window.addEventListener( 'resize', this.resizeListener );
-		}
+		debug( 'init', { maxRequestedPage, posts: this.props.posts } );
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		if ( ! isEqual( this.props.query, nextProps.query ) ) {
+		if (
+			! isEqual( this.props.query, nextProps.query ) ||
+			! isEqual( this.props.siteId, nextProps.siteId )
+		) {
+			const maxRequestedPage = this.estimatePageCountFromPosts( nextProps.posts );
 			this.setState( {
-				requestedPages: this.getInitialRequestedPages( nextProps )
+				maxRequestedPage,
 			} );
+			debug(
+				'reset maxRequestedPage to %d due to query change',
+				maxRequestedPage,
+				{
+					prevQuery: this.props.query,
+					nextQuery: nextProps.query,
+				}
+			);
 		}
-	}
 
-	componentWillUnmount() {
-		if ( this.resizeListener ) {
-			window.removeEventListener( 'resize', this.resizeListener );
-			delete this.resizeListener;
+		const prevLength = this.props.posts && this.props.posts.length;
+		const nextLength = nextProps.posts && nextProps.posts.length;
+		if ( prevLength !== nextLength ) {
+			debug( 'posts.length changed', {
+				prevLength,
+				nextLength,
+				maxRequestedPage: this.state.maxRequestedPage,
+				lastPage: nextProps.lastPage,
+			} );
```

Comment:
If we're going to merge this PR, we can definitely remove this block since it serves no purpose other than debug info.  I don't have a strong preference on whether we would remove the other `debug` statements.

Reply:
> If we're going to merge this PR, we can definitely remove this block

Shall we remove this block then before merge? And consider removing the other debug statements?